### PR TITLE
Add etag while fetching translations

### PR DESCRIPTION
**Pull request by candrikos**
We need to use etag policy while retrieving translations to eliminate performance issues related to duplicate fetching.

### CODE REVIEW

File: transifex/native/cds.py [hunk 1]
```diff
@@ -27,6 +27,19 @@
...
+
+    def __init__(self):
+        self._mem = {}
+
+    def set(self, key, value):
+        self._mem[key] = value
+
+    def get(self, key):
+        return self._mem.get(key, '')
+
```
Comment (candrikos) [inline transifex/native/cds.py:41]: Probably we need to a duration policy here

Comment (kbairak) [reply transifex/native/cds.py:41]: Add this suggestion as a comment so it outlives this PR.

File: transifex/native/cds.py [hunk 2]
```diff
@@ -27,6 +27,19 @@
 }
 
 
+def EtagStore(object):
```
Comment (kbairak) [inline transifex/native/cds.py:30]: I am rather fond of using python's `__g/setitem__` methods.

In fact, if we were doing YAGNI we could use `dict`s for `EtagStore` instances and when the time came to add more functionality, we could define a `EtagStore` class that would inherit from `collections.UserDict` and override the `__getitem__` and `__setitem__` methods.

AFAIK the rest of the company prefers enforcing this kind of structure early on so I am more than ok with the current implementation.

File: transifex/native/cds.py [hunk 3]
```diff
@@ -108,10 +122,15 @@ def fetch_translations(self, language_code=None):
 
             for language_code in set(languages) & \
                     set(self.configured_language_codes):
+
+                req_headers = {"Authorization": "Bearer {token}".format(
+                        token=self.token),
+                        "If-None-Match": self.etags.get(language_code, '')
```
Comment (kbairak) [inline transifex/native/cds.py:128]: `self.etags.get` accepts one argument. It already returns `''` as a fallback value so simply lose the second argument.

Comment (candrikos) [reply transifex/native/cds.py:131]: @kbairak that is true ...

File: transifex/native/cache.py [hunk 4]
```diff
@@ -60,11 +64,16 @@ def update(self, data):
         :param dict data: the data to use in the cache, formatted as
             explained in AbstractCache.update()
         """
-        for lang_code, translations in data.items():
-            self._translations_by_lang[lang_code] = {
-                'translations': translations,
-                'last_update': now()
-            }
+        for lang_code, data in data.items():
+            should_update, translations = data
```
[the remark this answers is not in the document]
Comment (candrikos) [reply transifex/native/cache.py:68]: For sure!

**Pull request merged**